server/backend/opensearch_security_client.ts

```diff
@@ -182,7 +193,9 @@ export class SecurityClient {
   public async getSamlHeader(request: OpenSearchDashboardsRequest) {
     try {
       // response is expected to be an error
-      await this.esClient.asScoped(request).callAsCurrentUser('opensearch_security.authinfo');
+      await this.esClient.asScoped(request).callAsCurrentUser('opensearch_security.authinfo', {
+        [AUTH_TYPE_PARAM]: AuthType.SAML,
```

Comment:
only this will work, as all the following requests will have Authorization header (from cookie), and if they don't the request will be redirected to login page. This is what my understanding is.

Reply:
Maybe I didn't get this change completely - shouldn't the backend only be checking for "anonymous"? Meaning this wouldn't have any effect, even without the authorization header?

Reply:
Specifically this block:

```
if (params.containsKey("auth_type")) {
            return params.get("auth_type").equals("anonymous");
        }
        return !headers.containsKey(HttpHeaders.AUTHORIZATION);
```

should return false even if SAML doesn't pass it in, right?

Reply:
`[AUTH_TYPE_PARAM]: AuthType.SAML` is required to differentiate the normal unauth request (anonymous) from SAML unauth request, since SAML expects backend to pass the redirect route. without this differentiation, SAML request would be considered as anonymous accidentally by backend while would lead to 500 being thrown.

Reply:
Code like this on the backend:

```
private boolean isRequestForAnonymousLogin(Map<String, String> params, Map<String, List<String>> headers) {
 helper = !headers.containsKey(HttpHeaders.AUTHORIZATION);
        if (params.containsKey("auth_type")) {
            return helper && params.get("auth_type").equals("anonymous");
        }
        return helper;
    }
    ```

Or its logical equivalent would make passing in SAML unecessary